Project_Starting_Point/Client/Client.cpp

```diff
@@ -21,39 +21,41 @@ int main()
 	ClientSocket = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
 	SvrAddr.sin_family = AF_INET;
 	SvrAddr.sin_port = htons(27001);
-	SvrAddr.sin_addr.s_addr = inet_addr("127.0.0.1");
-	connect(ClientSocket, (struct sockaddr*)&SvrAddr, sizeof(SvrAddr));
+	SvrAddr.sin_addr.s_addr = inet_addr("127.0.0.1"); 
+	connect(ClientSocket, (struct sockaddr*)&SvrAddr, sizeof(SvrAddr)); // connecting client with server
 
-	uiSize = GetSize();
-	for (unsigned int l = 0; l < uiSize; l++)
+	uiSize = GetSize(); // calling in-home function which returns size O(n)
+	for (unsigned int l = 0; l < uiSize; l++)  //O(n)
 	{
 		string strInput;
-		ifstream ifs("DataFile.txt");
-		for (unsigned int iStart = 0; iStart < l; iStart++)
+		ifstream ifs("DataFile.txt"); // opens ifstream object on File 
+		for (unsigned int iStart = 0; iStart < l; iStart++) // O(n) // this is costly and time consuming as it really will make it O(n^2) efficient
 			getline(ifs, strInput);
 
-		getline(ifs, strInput);
-		if (l > 0)
+		getline(ifs, strInput); 
+		if (l > 0) // if it's not the first line there are numbers of data following
 		{
 			size_t offset, preOffset;
 			offset = preOffset = 0;
 			unsigned int iParamIndex = 0;
 			//while (offset != std::string::npos)
-			while(iParamIndex != 8)
+			while(iParamIndex != 8) // go through until all parameters are calculated
 			{
 				offset = strInput.find_first_of(',', preOffset+1);
-				string strTx = strInput.substr(preOffset+1, offset - (preOffset+1));
-				send(ClientSocket, ParamNames[iParamIndex].c_str(), (int)ParamNames[iParamIndex].length(), 0);
+				string strTx = strInput.substr(preOffset+1, offset - (preOffset+1)); // this gets the string reading of the value: number 
+				send(ClientSocket, ParamNames[iParamIndex].c_str(), (int)ParamNames[iParamIndex].length(), 0); // this sends to server the name of parameter, as well as its length
 				recv(ClientSocket, Rx, sizeof(Rx), 0);
-				send(ClientSocket, strTx.c_str(), (int)strTx.length(), 0);
+				send(ClientSocket, strTx.c_str(), (int)strTx.length(), 0); // this now sends the string object that contains the number as a c string, passing its length as well
```

Comment:
Since this is a network transfer, we should be more specific. The c string is an array (so contiguous RAM) containing characters which are bytes encoded in ASCII.